src/acvp_kdf135_ikev2.c

```diff
@@ -35,12 +35,58 @@
  * Forward prototypes for local functions
  */
 static ACVP_RESULT acvp_kdf135_ikev2_output_tc (ACVP_CTX *ctx, ACVP_KDF135_IKEV2_TC *stc, JSON_Object *tc_rsp) {
-    json_object_set_string(tc_rsp, "sKeySeed", (const char *)stc->s_key_seed);
-    json_object_set_string(tc_rsp, "sKeySeedReKey", (const char *)stc->s_key_seed_rekey);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterial", (const char *)stc->derived_keying_material);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterialChild", (const char *)stc->derived_keying_material_child);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterialDh", (const char *)stc->derived_keying_material_child_dh);
-    return ACVP_SUCCESS;
+    ACVP_RESULT rv = ACVP_SUCCESS;
+    char *tmp = NULL;
+    
+    tmp = calloc(ACVP_KDF135_IKEV2_SKEY_SEED_STR_MAX+1, sizeof(char));
+    if (!tmp) { return ACVP_MALLOC_FAIL; }
+    
+    rv = acvp_bin_to_hexstr(stc->s_key_seed, stc->key_out_len, tmp);
```

Comment:
For the length check, what do you think about adding an extra parameter to acvp_bin_to_hexstr such as "dest_max", so that the function can handle "key_out_len" checking internally as it goes along incrementally during the conversion?

Reply:
Alternative is doing the length check before invoking acvp_bin_to_hexstr

Reply:
I think changing the signature would simplify things, good idea! Working through it now... there are a lot of references to that API :)